src/utils/WindowCloseWarnPopup.jsx

```diff
@@ -0,0 +1,26 @@
+import { Component } from 'react';
+
+class WindowCloseWarnPopup extends Component {
+  componentDidMount = () => {
+    window.addEventListener('beforeunload', this.keepOnPage);
+  };
+
+  componentWillUnmount = () => {
+    window.removeEventListener('beforeunload', this.keepOnPage);
+  };
+
+  keepOnPage = event => {
+    const message = 'Warning! Unsaved changes';
+
+    // eslint-disable-next-line no-param-reassign
+    event.returnValue = message;
+
+    return message;
+  };
+
+  render() {
+    return null;
+  }
+}
+
+export default WindowCloseWarnPopup;
```

Comment:
We could probably rewrite as:

```jsx
import React, { Component } from 'react';

// Ask the user for confirmation to leave the page.
// This is useful to use when a page has a form.
export default WrappedComponent =>
  class withAlertOnClose extends Component {
    componentDidMount = () => {
      // A generic string not under the control of the web page
      // will be shown instead of the returned string.
      // Custom text is not well supported yet.
      window.addEventListener('beforeunload', this.keepOnPage);
    };

    componentWillUnmount = () => {
      window.removeEventListener('beforeunload', this.keepOnPage);
    };

    keepOnPage = e => {
      e.preventDefault();
      e.returnValue = true;
    };

    render() {
      return <WrappedComponent {...this.props} />;
    }
  };

```